eval/mt_bench/components.py

```diff
@@ -0,0 +1,169 @@
+        import requests
+
+        command = f"nohup python3.11 -m vllm.entrypoints.openai.api_server --model {model_path} &"
+        subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        server_url = "http://localhost:8000/v1"
+        print(f"Waiting for vLLM server to start at {server_url}...")
+
+        for attempt in range(retries):
+            try:
+                response = requests.get(f"{server_url}/models")
+                if response.status_code == 200:
+                    print(f"vLLM server is up and running at {server_url}.")
+                    return
+            except requests.ConnectionError:
+                pass
+
+            print(f"Server not available yet, retrying in {delay} seconds (Attempt {attempt + 1}/{retries})...")
+            time.sleep(delay)
+        
+        raise RuntimeError(f"Failed to start vLLM server at {server_url} after {retries} retries.")
+
+    # This seems like excessive effort to stop the vllm process, but merely saving & killing the pid doesn't work
+    # Also, the base image does not include `pkill` cmd, so can't pkill -f vllm.entrypoints.openai.api_server either
+    def stop_vllm_server_by_name():
+        import psutil
+
+        for process in psutil.process_iter(attrs=["pid", "name", "cmdline"]):
+            cmdline = process.info.get("cmdline")
+            if cmdline and "vllm.entrypoints.openai.api_server" in cmdline:
+                print(f"Found vLLM server process with PID: {process.info['pid']}, terminating...")
+                try:
+                    process.terminate()  # Try graceful termination
+                    process.wait(timeout=5)  # Wait a bit for it to terminate
+                    if process.is_running():
+                        print(f"Forcefully killing vLLM server process with PID: {process.info['pid']}")
+                        process.kill()  # Force kill if it's still running
+                    print(f"Successfully stopped vLLM server with PID: {process.info['pid']}")
+                except psutil.NoSuchProcess:
+                    print(f"Process with PID {process.info['pid']} no longer exists.")
+                except psutil.AccessDenied:
+                    print(f"Access denied when trying to terminate process with PID {process.info['pid']}.")
+                except Exception as e:
+                    print(f"Failed to terminate process with PID {process.info['pid']}. Error: {e}")
+
+
+    import json
+    import torch
+    import os
+
+    from instructlab.eval import mt_bench_answers, mt_bench_judgment
+
+    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+    candidate_server_url = "http://localhost:8000/v1"
+
+    gpu_available = torch.cuda.is_available()
+    gpu_name = torch.cuda.get_device_name(torch.cuda.current_device()) if gpu_available else "No GPU available"
+
+    print(f"GPU Available: {gpu_available}, {gpu_name}")
+
+    if max_workers == "auto":
```

Comment:
Can you please document the magic const `auto`? It may be not obvious this comes hardcoded from ILab codebase.

Reply:
I thought I did that? Oh, I did that in mmlu/components

Reply:
added